File: 최승훈/2주차/10828.py
import sys

N = int(sys.stdin.readline())

stack = []

for _ in range(N):
    # stdin: input stream / readline(): reads a line / .split() : splits -> Strings
    word = sys.stdin.readline().split()
    command = word[0]
    
    # command: push
    if command == "push":
        value = word[1]
        stack.append(value)

    # command: pop
    elif command == "pop":
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())

    # command: size
    elif command == "size":
        print(len(stack))

    # command: empty
    elif command == "empty":
        if len(stack) == 0:
            print(1)
        else:
            print(0)

    # command: top
    elif command == "top":
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])

File: 권현욱/2주차/10828.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int N;
    cin >> N;

    stack<int> s;
    string str;

    for (int i = 0; i < N; i++)
    {
        cin >> str;

        if (str == "push")
        {
            int num;
            cin >> num;
            s.push(num);
        }
        else if (str == "pop")
        {
            if (s.empty())
                cout << "-1" << "\n";
            else 
            {
                cout << s.top() << "\n";
                s.pop();
            }
        }
        else if (str == "size")
            cout << s.size() << "\n";
        else if (str == "empty")
        {
            if (s.empty())
                cout << 1 << "\n";
            else
                cout << 0 << "\n";
        }
        else if (str == "top")
        {
            if (s.empty())
                cout << -1 << "\n";
            else
                cout << s.top() << "\n";
        }
    }

    return 0;
}
